Add SectionName Attribute

MikeCardiff opened the pull request:
Adds the `SectionName` attribute to specify the appsettings section to use if it is differs from the class name.

E.g.

```csharp
[SectionName("ConfirmationEmail")]
public class ConfirmationEmailSettingsFromAttribute : IAppSettings
{
	public required string Subject { get; set; }
}
```

Commit:
Add SectionName Attribute and examples

Commit:
Add nested class library example

IeuanWalker commented:
@MikeCardiff sorry, can you update your PR based on the latest changes.

Main change that effects this PR, is it no longer gets the section name at runtime, it sets it in the source generator. So this method would need to be updated to look for the attribute rather than the property - https://github.com/IeuanWalker/AppSettings/blob/0f11a3291ee169555a357c99ede33ef869b126c8/src/IeuanWalker.AppSettings.Generator/AppSettingsGenerator.cs#L144

Also whilst your at it can you remove the property from the interface, and we'll just use the attribute

______

Let me know if you want me to look at it instead

Commit:
Merge branch 'master' into sectionNameAttribute

# Conflicts:
#	Examples/ApiProject/ExampleAppSettings.cs
#	Examples/ApiProject/Program.cs
#	Examples/BlazorProject/Components/Pages/Home.razor
#	Examples/BlazorProject/Components/Pages/Home.razor.cs
#	Examples/BlazorProject/ExampleAppSettings.cs
#	Examples/ConsoleProject/ExampleAppSettings.cs
#	Examples/ConsoleProject/Program.cs
#	src/IeuanWalker.AppSettings/AppSettingsExtensions.cs

Commit:
Merge latest changes and retrieve SectionName attribute in source generator. Also update the example settings and readme.

Commit:
cleanup

MikeCardiff commented:
@IeuanWalker Updated with the latest changes from master and moved the attribute check to the source generator. Also removed the SectionName property form the interface.

IeuanWalker committed:
Updated to be type safe

IeuanWalker committed:
Update documentation in SectionNameAttribute.cs

Enhanced comments for clarity on attribute usage,
including support for hierarchical paths and corrected
example syntax.

IeuanWalker reviewed (approved):
LGTM, thanks :)
